src/ComputeSharp.D2D1/Shaders/Interop/Helpers/Rectangle64.cs

```diff
@@ -0,0 +1,159 @@
+    /// </summary>
+    /// <param name="x">The left coordinate of the rectangle.</param>
+    /// <param name="y">The top coordinate of the rectangle.</param>
+    /// <param name="width">The width of the rectangle.</param>
+    /// <param name="height">The height of the rectangle.</param>
+    public Rectangle64(long x, long y, long width, long height)
+    {
+        this.x = x;
+        this.y = y;
+        this.width = width;
+        this.height = height;
+    }
+
+    /// <summary>
+    /// Gets the left coordinate of the current rectangle.
+    /// </summary>
+    public readonly long Left
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        get => this.x;
+    }
+
+    /// <summary>
+    /// Gets the top coordinate of the current rectangle.
+    /// </summary>
+    public readonly long Top
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        get => this.y;
+    }
+
+    /// <summary>
+    /// Gets the right coordinate of the current rectangle.
+    /// </summary>
+    public readonly long Right
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        get => this.x + this.width;
+    }
+
+    /// <summary>
+    /// Gets the bottom coordinate of the current rectangle.
+    /// </summary>
+    public readonly long Bottom
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        get => this.y + this.height;
+    }
+
+    /// <summary>
+    /// Gets a <see cref="Rectangle64"/> instance representing an empty rectangle.
+    /// </summary>
+    /// <remarks>The returned instance is such that computing the union of it with any othe rectangle yields the other rectangle.</remarks>
+    public static Rectangle64 Empty => new(long.MaxValue, long.MaxValue, long.MinValue, long.MinValue);
+
+    /// <summary>
+    /// Updates the current rectangle to be the union of itself and a given input rectangle.
+    /// </summary>
+    /// <param name="rectangle">The input <see cref="Rectangle64"/> instance to compute the union with.</param>
+    public void Union(Rectangle64 rectangle)
+    {
+        long left = Math.Min(this.Left, rectangle.Left);
+        long top = Math.Min(this.Top, rectangle.Top);
+        long right = Math.Max(this.Right, rectangle.Right);
+        long bottom = Math.Max(this.Bottom, rectangle.Bottom);
+
+        this.x = left;
+        this.y = top;
+        this.width = right - left;
+        this.height = bottom - top;
+    }
+
+    /// <summary>
+    /// Updates the current rectangle to be the intersection of itself and a given input rectangle.
+    /// </summary>
+    /// <param name="rectangle">The input <see cref="Rectangle64"/> instance to compute the intersection with.</param>
+    public void Intersect(Rectangle64 rectangle)
+    {
+        long maxLeft = Math.Max(this.x, rectangle.x);
+        long maxTop = Math.Max(this.y, rectangle.y);
+        long width = Math.Max(Math.Min(this.Right, rectangle.Right) - maxLeft, 0);
```

Comment:
probably easier to read if you compute `maxRight` and `maxBottom`, _then_ calculate `width` and `height`

Reply:
Yeah good idea. Updated in f7b44531533729c1f70f2c429143bb484cf72b10, made it consistent to `Union` 🙂